cmd/models/certificate.model.go

```diff
@@ -0,0 +1,84 @@
+package models
+
+import (
+	"go-lms-of-pupilfirst/pkg/utils"
+)
+
+var (
+	certificateTableName = "certificates"
+)
+
+// Certificate defines a model for student certificates in a model
+type Certificate struct {
+	utils.Base
+	CourseID      string `sql:"type:uuid;" validate:"omitempty,uuid,required"`
+	IssuerID      string `sql:"type:uuid;" validate:"omitempty,uuid,required"`
+	QRCorner      string
+	QRScale       int
+	Margin        int
+	NameOffsetTop int
+	FontSize      int
+	Message       string
+	Active        bool
+
+	Course             *Course               `gorm:"foreignkey:CourseID"`
+	Issuer             *CourseAuthor         `gorm:"foreignkey:IssuerID"`
+	IssuedCertificates IssuedCertificateList `gorm:"foreignkey:CertificateID"`
```

Comment:
Commenting for here and everywhere else you have these foreign key relations. 

Have you tested that this works? Is gorm automatically doing the join for you here? I've never seen the DB structs written this way.

Reply:
I have not started testing (yet). But I reckon that the tags will help gorm build a relationship when performing ORM actions like `Related` function.

This does not create actual relationship in the DB tho.

Reply:
I would definitely try writing some tests to make sure this works first. You may find that this doesn't work.

Reply:
alright, I will do that.